File: Python/pointer/remove_duplicates_from_sorted_array_ii.py
from typing import *

def removeDuplicates(nums: List[int]) -> int:
    length = len(nums)
    if length <= 2: return length
    slow, fast = 2, 2
    while fast < length:
        if nums[fast] != nums[slow - 2]:
            nums[slow] = nums[fast]
            slow += 1
        fast += 1
    
    return slow

File: core/tmp/Java/src/com/dbc/algorithm/Pointer/RemoveDuplicatesFromSortedArrayII.java
package com.dbc.algorithm.Pointer;

public class RemoveDuplicatesFromSortedArrayII {
    public int removeDuplicates(int[] nums) {
        int length = nums.length;
        if (length <= 2) return length;
        int slow = 2, fast = 2;
        while (fast < length) {
            if (nums[fast] != nums[slow - 2]) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        return slow;
    }
}
